paddle/fluid/operators/sum_op.cc

```diff
@@ -111,7 +111,7 @@ class SumOp : public framework::OperatorWithKernel {
                        "Input var[%s] should not be nullptr", x_vars_name[idx]);
         auto tensor =
             framework::GetLoDTensorOrSelectedRowsValueFromVar(*x_vars[idx]);
-        if (tensor->numel() == 0) {
+        if (tensor->numel() <= 0 || (!tensor->IsInitialized())) {
```

Comment:
Is that a better way to prevent the non-initialized tensors? It is really a hard problem... But it may be not suitable to hide problems in codes.

Reply:
Actually, in GAN model with double backward, there is non-initialized tensors for sum op. Since the double_grad ops (see conv_double_grad and mul_double_grad) does not add the var in op_desc, it is None.   We can run without this change.

Here, change == 0  to (<=0 || ! IsInitialized()) is more rigorous.